diff --git a/public/assets/css/modules/challenges.css b/public/assets/css/modules/challenges.css
new file mode 100644
index 0000000..492135c
--- /dev/null
+++ b/public/assets/css/modules/challenges.css
@@ -0,0 +1,4 @@
+
+.challenges {
+  display:none;
+}
diff --git a/public/assets/js/styleguide.js b/public/assets/js/styleguide.js
index 3ead576..e8e20f3 100644
--- a/public/assets/js/styleguide.js
+++ b/public/assets/js/styleguide.js
@@ -40,7 +40,24 @@ var createDivError = function(){
   document.getElementById('iframe--container').appendChild(divSmsError);
 }
 
+var getChallenges = function(event){
+  var id = this.href.substring(this.href.lastIndexOf('#'));
+  var challengeContent = document.querySelector(id + ' .challenges').innerHTML;
+  setChallenges(challengeContent);
+}
+
+var setChallenges = function(content){
+  debugger
+  editor.setValue(content);
+  ajax();
+}
+
 document.addEventListener('DOMContentLoaded', function() {
   ajax();
   document.getElementById('formArea').addEventListener('submit', generateStyleGuide);
-});
\ No newline at end of file
+  var all = document.querySelectorAll('.challenges_examples .anchor');
+  for ( var i = 0; i < all.length; i++) {
+    all[i].addEventListener('click', getChallenges);
+  }
+
+});
diff --git a/views/challenges.erb b/views/challenges.erb
index 03147db..69abf0b 100644
--- a/views/challenges.erb
+++ b/views/challenges.erb
@@ -1,26 +1,16 @@
 <div id="content">
   <ul id="tabs" class="nav nav-tabs" data-tabs="tabs">
-    <li class="active"><a class="anchor" href="#c-colors" data-toggle="tab">Colors</a></li>
-    <li><a class="anchor" href="#c-font" data-toggle="tab">FONT</a></li>
-    <li><a class="anchor" href="#c-html" data-toggle="tab">HTML</a></li>
-    <li><a class="anchor" href="#c-javascript" data-toggle="tab">Javascript</a></li>
-    <li><a class="anchor" href="#c-filter" data-toggle="tab">Other Filter</a></li>
+    <li class="active"><a class="anchor" href="#c-colors" data-toggle="tab">.</a></li>
+    <li><a class="anchor" href="#c-font" data-toggle="tab">.</a></li>
+    <li><a class="anchor" href="#c-html-plain" data-toggle="tab">.</a></li>
+    <li><a class="anchor" href="#c-javascript" data-toggle="tab">.</a></li>
+    <li><a class="anchor" href="#c-search" data-toggle="tab">.</a></li>
   </ul>
   <div id="my-tab-content" class="tab-content">
-    <div class="tab-pane active" id="c-colors">
-        <h5>Colors</h5>
-    </div>
-    <div class="tab-pane" id="c-font">
-        <h5>FONT</h5>
-    </div>
-    <div class="tab-pane" id="c-html">
-        <h5>HTML</h5>
-    </div>
-    <div class="tab-pane" id="c-javascript">
-        <h5>Javascript</h5>
-    </div>
-    <div class="tab-pane" id="c-filter">
-        <h5>Other Filters</h5>
-    </div>
+    <div class="tab-pane active" id="c-colors"><%= erb :colors, views: 'views/challenges_examples' %></div>
+    <div class="tab-pane" id="c-font"><%= erb :font, views: 'views/challenges_examples' %></div>
+    <div class="tab-pane" id="c-html-plain"><%= erb :html, views: 'views/challenges_examples' %></div>
+    <div class="tab-pane" id="c-javascript"><%= erb :javascript, views: 'views/challenges_examples' %></div>
+    <div class="tab-pane" id="c-search"><%= erb :search, views: 'views/challenges_examples' %></div>
   </div>
 </div>
diff --git a/views/challenges_examples/colors.erb b/views/challenges_examples/colors.erb
new file mode 100644
index 0000000..9ec4def
--- /dev/null
+++ b/views/challenges_examples/colors.erb
@@ -0,0 +1,16 @@
+<h5>Colors</h5>
+<p>
+</br>
+Play with dashes(-) to add emphy spaces </br>
+</p>
+<pre><code>
+- -       -       -
+- - #cc2301 #b1ff91 #a3ae9e
+- - #991a00 #7bcb5a #465141
+</pre></code>
+
+<div class="challenges">
+@colors:
+#cc2301 #b1ff91 #a3ae9e
+#991a00 #7bcb5a #465141
+</div>
diff --git a/views/challenges_examples/font.erb b/views/challenges_examples/font.erb
new file mode 100644
index 0000000..680b640
--- /dev/null
+++ b/views/challenges_examples/font.erb
@@ -0,0 +1,12 @@
+
+<h5>Font</h5>
+<p>
+  Add a link to a font from Internet before your example to load it:
+  <pre><code>
+<%= CGI.escapeHTML "<link href='https://fonts.googleapis.com/css?family=Indie+Flower' rel='stylesheet' type='text/css'>"%>
+  </pre></code>
+</p>
+<div class="challenges">
+@font-example 20px Indie Flower:
+	My example font
+</div>
diff --git a/views/challenges_examples/html.erb b/views/challenges_examples/html.erb
new file mode 100644
index 0000000..13466cf
--- /dev/null
+++ b/views/challenges_examples/html.erb
@@ -0,0 +1,14 @@
+<h5>HTML</h5>
+<p>
+</br>
+Try this:</br>
+To output only the content in a code-block specify plain as the code block language
+</p>
+<div class="challenges">
+```
+<div class="block">
+  <h1 class="title">Extract content</h1>
+  <p> This is a content inside of a code block</p>
+</div>
+```
+</div>
diff --git a/views/challenges_examples/javascript.erb b/views/challenges_examples/javascript.erb
new file mode 100644
index 0000000..7581b65
--- /dev/null
+++ b/views/challenges_examples/javascript.erb
@@ -0,0 +1,25 @@
+<h5>Javascript</h5>
+<p>
+</br>
+  If you want run JQUERY only add the link before the javascript block:
+</p>
+<pre><code>
+<%= CGI.escapeHTML '<script src="http://code.jquery.com/jquery-2.1.4.min.js" type="text/javascript"></script>' %>
+</pre></code>
+<div class="challenges">
+
+```plain
+<button>If you click on me, I will disappear.</button>
+
+
+<button>If you click on me, I will disappear.</button>
+```
+```javascript
+@javascript
+$(document).ready(function(){
+    $("button").click(function(){
+        $(this).hide();
+    });
+});
+```
+</div>
diff --git a/views/challenges_examples/search.erb b/views/challenges_examples/search.erb
new file mode 100644
index 0000000..821615e
--- /dev/null
+++ b/views/challenges_examples/search.erb
@@ -0,0 +1,52 @@
+<h5>Search</h5>
+<p>
+  Now that you already tried a few examples you may want to add a search bar to filter your examples:
+<pre><code>
+  @search-box
+</pre></code>
+ You could even change the default placeholder:
+ <pre><code>
+   @search-box placeholder: buscar...
+ </pre></code>
+</p>
+<div class="challenges">
+<script src="http://code.jquery.com/jquery-2.1.4.min.js" type="text/javascript"></script>
+<link href='https://fonts.googleapis.com/css?family=Indie+Flower' rel='stylesheet' type='text/css'>
+# Colors
+
+@colors:
+- -       -       -
+- - #cc2301 #b1ff91 #a3ae9e
+- - #991a00 #7bcb5a #465141
+
+# font
+
+@font-example 20px Indie Flower:
+	My example font
+
+# html
+
+```
+<div class="block">
+  <h1 class="title">Extract content</h1>
+  <p> This is a content inside of a code block</p>
+</div>
+```
+
+# javascript
+
+```plain
+<button>If you click on me, I will disappear.</button>
+
+
+<button>If you click on me, I will disappear.</button>
+```
+```javascript
+@javascript
+$(document).ready(function(){
+    $("button").click(function(){
+        $(this).hide();
+    });
+});
+```
+</div>
diff --git a/views/index.erb b/views/index.erb
index 89b1a3c..6fcc8af 100644
--- a/views/index.erb
+++ b/views/index.erb
@@ -5,6 +5,7 @@
   <link rel="stylesheet" type="text/css" href="assets/css/modules/button--generate-style-guide.css">
   <link rel="stylesheet" type="text/css" href="assets/css/modules/header.css">
   <link rel="stylesheet" type="text/css" href="assets/css/modules/documentation-menu.css">
+  <link rel="stylesheet" type="text/css" href="assets/css/modules/challenges.css">
   <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.4.0/css/font-awesome.min.css">
   <script type="text/javascript" src= "assets/js/ace/ace.js"></script>
 </head>
@@ -43,7 +44,7 @@
 
             <input type="checkbox" class="help-toggle" id="challenges-button"></input>
             <label class='help-challenges' for="challenges-button"><i class="fa fa-star"></i></label>
-            <div class="documentation">
+            <div class="documentation challenges_examples">
               <%= erb :challenges %>
             </div>
 
@@ -65,4 +66,4 @@
   <script src="assets/js/bootstrap.min.js"></script>
 
 </body>
-</html>
\ No newline at end of file
+</html>
